File: pg-common/src/pg/const.clj
(ns pg.const
  (:import
   java.time.Duration
   java.time.Instant
   java.time.LocalDate
   java.time.ZoneOffset))


(def ^Duration PG_EPOCH_DIFF
  (Duration/between Instant/EPOCH
                    (-> (LocalDate/of 2000 1 1)
                        (.atStartOfDay)
                        (.toInstant ZoneOffset/UTC))))


(def MICROS 1000000)

(def MILLIS 1000)

(def PROTOCOL_VERSION 196608)

(def SOURCE_PORTAL \P)
(def SOURCE_STATEMENT \S)

(def FORMAT_TXT 0)
(def FORMAT_BIN 1)

(def CANCEL_CODE 80877102)

(def TX_ERROR :E)
(def TX_TRANSACTION :T)
(def TX_IDLE :I)

(def SSL_CODE 80877103)
(def SSL_YES \S)
(def SSL_NO \N)

(def SCRAM_SHA_256 "SCRAM-SHA-256")

(def NUMERIC_POS (int 0x0000))
(def NUMERIC_NEG (int 0x4000))
(def NUMERIC_NAN (int 0xC000))

(def COPY_BUFFER_SIZE 2048)

(def COPY_FAIL_MSG "COPY has been interrupted by the client")

(def COPY_CSV_CELL_SEP \,)
(def COPY_CSV_LINE_SEP "\r\n")
(def COPY_CSV_NULL "")

(def COPY_FORMAT_TXT :txt)
(def COPY_FORMAT_CSV :csv)
(def COPY_FORMAT_BIN :bin)

(def IN_STREAM_BUF_SIZE 0xFFFF)

File: pg-client2/src/com/github/igrishaev/Const.java
package com.github.igrishaev;

public class Const {
    public static final int PROTOCOL_VERSION = 196608;
    public static final int CANCEL_CODE = 80877102;
    public static final int SSL_CODE = 80877103;
    public static final int COPY_BUFFER_SIZE = 2048;
    public static final String COPY_FAIL_MSG = "COPY has been interrupted by the client";
    public static final int PG_PORT = 5432;
    public static final String PG_HOST = "127.0.0.1";
    public static final int IN_STREAM_BUF_SIZE = 0xFFFF;
    public static final int OUT_STREAM_BUF_SIZE = 0xFFFF;
    public static final String UTF8 = "UTF-8";
    public static final long EXE_MAX_ROWS = 0xFFFFFFFFL;
    public static final int JSON_ENC_BUF_SIZE = 256;
    public static final String APP_NAME = "pg2";
    public static final String CLIENT_ENCODING = "UTF8";
    public static final char NULL_TAG = (char) 0;
    public static final String COPY_CSV_NULL = "";
    public static final String COPY_CSV_CELL_SEP = ",";
    public static final String COPY_CSV_CELL_QUOTE = "\"";
    public static final String COPY_CSV_LINE_SEP = "\r\n";
    public static final byte[] COPY_BIN_HEADER = {
            // header
            (byte) 'P',
            (byte) 'G',
            (byte) 'C',
            (byte) 'O',
            (byte) 'P',
            (byte) 'Y',
            (byte) 10,
            (byte) 0xFF,
            (byte) 13,
            (byte) 10,
            (byte) 0,
            // 0 int32
            (byte) 0, (byte) 0, (byte) 0, (byte) 0,
            // 0 int32
            (byte) 0, (byte) 0, (byte) 0, (byte) 0
    };
    public static final byte[] shortMinusOne = new byte[] {(byte) -1, (byte) -1};
}
